src/category_theory/sites/dense_subsite.lean

```diff
@@ -411,6 +416,68 @@ begin
   apply sheaf_hom_eq
 end
 
+/-- A fully faithful cover-dense functor preserves compatible families. -/
+lemma compatible_preserving [faithful G] : compatible_preserving K G :=
+begin
+  split,
+  intros ℱ Z T x hx Y₁ Y₂ X f₁ f₂ g₁ g₂ hg₁ hg₂ eq,
+  apply H.ext,
+  intros W i,
+  simp only [← functor_to_types.map_comp_apply, ← op_comp],
+  rw ← G.image_preimage (i ≫ f₁),
+  rw ← G.image_preimage (i ≫ f₂),
+  apply hx,
+  apply G.map_injective,
+  simp[eq]
+end
+
+noncomputable
+instance sites.pullback.full [faithful G] (Hp : cover_preserving J K G) :
+  full (sites.pullback A H.compatible_preserving Hp) :=
+{ preimage := λ ℱ ℱ' α, H.sheaf_hom α,
+  witness' := λ ℱ ℱ' α, H.sheaf_hom_restrict_eq α }
+
+instance sites.pullback.faithful [faithful G] (Hp : cover_preserving J K G) :
+  faithful (sites.pullback A H.compatible_preserving Hp) :=
+{ map_injective' := λ ℱ ℱ' α β (eq : whisker_left G.op α = whisker_left G.op β),
+  by rw [← H.sheaf_hom_eq α, ← H.sheaf_hom_eq β, eq] }
+
 end cover_dense
 
 end category_theory
+
+namespace category_theory.cover_dense
+
+open category_theory
+
+variables {C : Type u} [small_category C] {D : Type u} [small_category D]
+variables {G : C ⥤ D} [full G] [faithful G]
+variables {J : grothendieck_topology C} {K : grothendieck_topology D}
+variables {A : Type v} [category.{u} A] [limits.has_limits A]
+variables (Hd : cover_dense K G) (Hp : cover_preserving J K G) (Hl : cover_lifting J K G)
+
+include Hd Hp Hl
+
+/--
+Given a functor between small sites that is cover-dense, cover-preserving, and cover-lifting,
+it induces an equivalence of category of sheaves valued in a complete category.
+-/
+noncomputable
+def Sheaf_equiv_of_cover_preserving_cover_lifting : Sheaf J A ≌ Sheaf K A :=
```

Comment:
It seems to me that it will be hard to unfold this definition. Or at least, the result will be somewhat unpleasant because of the `let α :=`.
Can you prove the two `simp`-lemmas

* `Sheaf_equiv_of_cover_preserving_cover_lifting_functor`
* `Sheaf_equiv_of_cover_preserving_cover_lifting_inverse`

Hopefully the proof is just `rfl`.

Reply:
`@[simps functor inverse]` generates them just fine. 
That said, these are equalities of functors. Do we want these as simp lemmas?

Reply:
I think we still want them. They are "equation lemmas" that explain how to unfold a definition. That way `dsimp` can clean up the goal (it will use `simp`-lemmas that are proven by `rfl`).